Build failure: cmake on mipsel

### Steps To Reproduce

    nix-build . -A pkgsCross.mipsel-linux-gnu.cmake

### Build log

```
/nix/store/in6kl3y5kygnk177f6g3yip7pc4sab2s-mipsel-unknown-linux-gnu-binutils-2.40/bin/mipsel-unknown-linux-gnu-ld: ../Utilities/cmcppdap/libcmcppdap.a(typeof.cpp.o): undefined reference to symbol '__atomic_fetch_sub_8@@LIBATOMIC_1.0'
/nix/store/in6kl3y5kygnk177f6g3yip7pc4sab2s-mipsel-unknown-linux-gnu-binutils-2.40/bin/mipsel-unknown-linux-gnu-ld: /nix/store/sz4kbwqdgxlwxcqg053qzangdmlxd6k9-mipsel-unknown-linux-gnu-gcc-13.2.0-lib/mipsel-unknown-linux-gnu/lib/libatomic.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

### Additional context

It seems this problem was introduced in the `3.26.4 -> 3.27.7` update in cc9458e2b9a56c085a34e3b648300e46a87e94a3

### Notify maintainers

@ttuegel @LnL7 @AndersonTorres 

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/cm/cmake/package.nix
+++ b/pkgs/by-name/cm/cmake/package.nix
@@ -110,7 +110,7 @@ stdenv.mkDerivation (finalAttrs: {
       --subst-var-by libc_dev ${lib.getDev stdenv.cc.libc} \
       --subst-var-by libc_lib ${lib.getLib stdenv.cc.libc}
     # CC_FOR_BUILD and CXX_FOR_BUILD are used to bootstrap cmake
-    configureFlags="--parallel=''${NIX_BUILD_CORES:-1} CC=$CC_FOR_BUILD CXX=$CXX_FOR_BUILD $configureFlags"
+    configureFlags="--parallel=''${NIX_BUILD_CORES:-1} CC=$CC_FOR_BUILD CXX=$CXX_FOR_BUILD $configureFlags $cmakeFlags"
   '';
 
   # The configuration script is not autoconf-based, although being similar;

